src/core/image.rs

```diff
@@ -89,6 +83,114 @@ where
     }
 }
 
+#[derive(Debug)]
+pub struct RunnableImage<I: Image> {
+    image: I,
+    image_args: I::Args,
+    image_tag: Option<String>,
+    name: Option<String>,
+    network: Option<String>,
+    env_vars: BTreeMap<String, String>,
+    volumes: BTreeMap<String, String>,
+    ports: Option<Vec<Port>>,
+}
+
+impl<I: Image> RunnableImage<I> {
+    pub fn image(&self) -> &I {
+        &self.image
+    }
+
+    pub fn image_args(&self) -> &I::Args {
+        &self.image_args
+    }
+
+    pub fn network(&self) -> &Option<String> {
+        &self.network
+    }
+
+    pub fn name(&self) -> &Option<String> {
+        &self.name
+    }
+
+    pub fn env_vars(&self) -> &BTreeMap<String, String> {
+        &self.env_vars
+    }
+
+    pub fn volumes(&self) -> &BTreeMap<String, String> {
+        &self.volumes
+    }
+
+    pub fn ports(&self) -> &Option<Vec<Port>> {
+        &self.ports
+    }
+}
+
+impl<I: Image> RunnableImage<I> {
+    pub fn with_image_args(self, args: <I as Image>::Args) -> Self {
+        Self {
+            image_args: args,
+            ..self
+        }
+    }
```

Comment:
I think we should change this into a constructor that also allows to pass in the `Image`:

```rust
    pub fn with_args(image: I, args: <I as Image>::Args) -> Self {
        Self {
			image: I,
            image_args: args,
            ..self
        }
    }
```

That is useful for cases where neither `Image`, nor `Image::Args` can implement `Default`.

Reply:
I think we can just use `RunnableImage::from((Image, Image::Args))` for this case? and avoid introducing `with_args()`?